Holo-Host/hp-admin
# 2020 06 16/incomplete offers

### Updates :
- manages new event state in actionable promises, as provided from DNA
- adds counter and functionality to trigger accepting of incomplete invoiced offers
- adds user message to display if timeout error occurs upon accepting an offer

---

## To Test:
- Run Agent 1: `npm run start:holofuel-ui-1` and  `start:holofuel-conductor-1`
- Run Agent 2: `npm run start:holofuel-ui-2` and  `start:holofuel-conductor-2`
- Create Offer from Agent 1 to Agent 2
- Have Agent 2 accept the offer, and immediately thereafter kill the running sim2h
- View flash message stating there was an internal timeout error
- Notice accepted offer in the Inbox is marked with `acceptance pending`
- Open the console and watch for 10 `list_pending` calls to queue up
- View flash message about internal timeout error
- Start sim2h again
- Notice `acceptOffer` mutation (which calls the `receive_payments_pending` zome call) to successfully process and handle the in-process offer
- Watch for the offer to disappear and flash message notifying a successful acceptance 

>NB: If you see a console error that states 'spender chain invalid` then you'll see a flash message stating that this transaction is now stale. If this happens, there was an internal error and you'll need to refresh your storage and start the test anew.
